core/services/flux_monitor.go

```diff
@@ -177,10 +179,11 @@ func (fm *concreteFluxMonitor) addAction(ctx context.Context, connected bool, jo
 			return errors.Wrap(err, "factory unable to create checker")
 		}
 		if connected {
```

Comment:
Why would we connect again if we are already connected?

Reply:
I'm not sure to be honest.  This change preserves the same behavior though, so I assume its intended.